pkg/apis/admissionregistration/types.go

```diff
@@ -256,6 +256,16 @@ type Validation struct {
 	// If not set, StatusReasonInvalid is used in the response to the client.
 	// +optional
 	Reason *metav1.StatusReason
+	// MessageExpression is an expression that compose a custom message to be
+	// displayed when the validation fails. The expression must evaluate to a
+	// string.
+	// If both MessageExpression and Message were set, the evaluation result of
+	// MessageExpression takes precedence if and only if the evaluation succeeds.
+	// If the evaluation fails for whatever reason, Message will be displayed instead.
+	// MessageExpression has access to the following variables:
+	// object, oldObject, params, and request
+	// +optional
```

Comment:
Noted. Will sync with said PR.

Reply:
Sorry forgot to mention that we'd better use the serialized field names in godoc which is `messageExpression` and also we will check if line breaks in messageExpression. Here is the updated doc:
```
// messageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
	// Since messageExpression is used as a failure message, it must evaluate to a string.
	// If both message and messageExpression are present on a validation, then messageExpression will be used if validation fails. 
	// If messageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced
	// as if the messageExpression field were unset. If messageExpression evaluates to an empty string, a string with only spaces, or a string
	// that contains line breaks, then the validation failure message will also be produced as if the messageExpression field were unset, and
	// the fact that messageExpression produced an empty string/string with only spaces/string with line breaks will be logged.
	// messageExpression has access to all the same variables as the `expression` except for 'authorizer' and 'authorizer.requestResource'.
	// Example:
	// "object.x must be less than max ("+string(self.max)+")"
	// +optional
```
So we have to check the linebreaks in expressionExpr as what we did here:
https://github.com/kubernetes/kubernetes/blob/c80a93d1a9fb62f62acd923be63124116bde3f58/pkg/apis/admissionregistration/validation/validation.go#L808-L809
Inside the same func, check the messageExpression successfully compiled as well

Reply:
Since, #115969 is merged, I am resolving this thread after last sync.